osp/core/session/sparql_backend.py

```diff
@@ -0,0 +1,89 @@
+"""Defines an abstract base class for backends that support SPARQL queries."""
+from abc import ABC, abstractmethod
+import uuid
+from osp.core.utils.general import iri_from_uid, uid_from_iri
+from osp.core.utils.general import CUDS_IRI_PREFIX
+
+
+class SPARQLBackend(ABC):
+    """Defines an abstract base class for backends that support SPARQL queries.
+
+    Contains only one abstract method, it is feasible to use multiple
+    inheritance with this abstract class.
+    """
+
+    def sparql(self, query_string):
+        """Execute the given SPARQL query on the backend.
+
+        Args:
+            query_string (): The SPARQL query as a string.
+        """
+        return self._sparql(query_string=query_string.replace(
```

Comment:
Maybe add a note for future us expaining why we need to replace the root's uid?

Reply:
Done.